app/models/mixins/process_tasks_mixin.rb

```diff
@@ -51,6 +49,48 @@ def invoke_tasks_local(options)
       end
     end
 
+    def invoke_tasks_remote(options)
+      ids_by_region = options[:ids].group_by { |id| ApplicationRecord.id_to_region(id.to_i) }
+      ids_by_region.each do |region, ids|
+        remote_options = options.merge(:ids => ids)
+        hostname = MiqRegion.find_by_region(region).remote_ws_address
+        if hostname.nil?
+          $log.error("An error occurred while invoking remote tasks...The remote region [#{region}] does not have a web service address.")
+          next
+        end
+
+        begin
+          invoke_api_tasks(hostname, remote_options)
+        rescue => err
+          # Handle specific error case, until we can figure out how it occurs
+          if err.class == ArgumentError && err.message == "cannot interpret as DNS name: nil"
+            $log.error("An error occurred while invoking remote tasks...")
+            $log.log_backtrace(err)
+            next
+          end
+
+          $log.error("An error occurred while invoking remote tasks...Requeueing for 1 minute from now.")
+          $log.log_backtrace(err)
+          MiqQueue.put(
+            :class_name  => base_class.name,
+            :method_name => 'invoke_tasks_remote',
+            :args        => [remote_options],
+            :deliver_on  => Time.now.utc + 1.minute
+          )
+          next
+        end
+
+        msg = "'#{options[:task]}' successfully initiated for remote VMs: #{ids.sort.inspect}"
+        task_audit_event(:success, options, :message => msg)
+      end
+    end
+
+    # This should be overridden by any class including this module
+    # and expecting to be asked to invoke tasks for a remote region
+    def invoke_api_tasks(_hostname, _remote_options)
+      raise NotImplementedError
```

Comment:
Okay, I split the commit to try to make this more clear.
f75744befd32f8b61b70fde674ce43ea0dae3636 has the initial move and 0711b24b594024e0eeec750171a06714d579ffc3 has the new method, the caller change and the SOAP change.

My `git log` shows those two commits as next to each other, but github seems to be ordering them differently for some reason. Looks like it should be find either way though.